src/client/consumer.ts

```diff
@@ -70,28 +73,40 @@ class Consumer implements vscode.Disposable {
      * Received messages and/or errors are emitted via events.
      */
     async start(): Promise<void> {
+        const partition = this.options.partition;
+        const fromOffset = this.options.fromOffset;
+        const topic = this.options.topic;
         this.kafkaClient = createKafka(this.options);
-        this.consumer = this.kafkaClient.consumer({ groupId: `vscode-kafka-${this.options.clusterId}-${this.options.topic}`, retry: { retries: 3 }});
+        this.consumer = this.kafkaClient.consumer({ groupId: this.options.consumerGroupId, retry: { retries: 3 } });
         await this.consumer.connect();
-        await this.consumer.subscribe({ topic: this.options.topic, fromBeginning: this.options.fromOffset ===  "earliest" })
+
+        const fromBeginning = fromOffset === "earliest" ? true : (fromOffset === "latest" ? false : undefined);
+        const subscribeOptions = (fromBeginning != undefined) ? { topic, fromBeginning } : { topic };
+        await this.consumer.subscribe(subscribeOptions)
 
         await this.consumer.run({
```

Comment:
https://kafka.js.org/docs/consuming#a-name-seek-a-seek says "you don't need to await consumer#run"

Reply:
Indeed I read that too, I believed that I did that. Now fixed.